### PR TITLE
Admin Menu: Centralize class loading in Jetpack codebase

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request:
As agreed on paYJgx-1t4-p2, and, in order to simplify the development experience, Jetpack should be the single source of truth for all the Nav Unification logic.

To that end, this PR adds the admin menu class for P2 sites that currently lives in the WP.com codebase. From now on, changes to that menu should be done in Jetpack. cc @Automattic/lighthouse 

This PR also adds the WP.com logic for determining which admin menu class should be instantiated, so we can re-use it in WP.com as well (see D59179-code). Furthermore, this class is filterable through the new `jetpack_admin_menu_class` filter (useful for Atomic hotfixes if we want to override the admin menu behavior).

#### Jetpack product discussion
paYJgx-1t4-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
This PR does not changes the current behavior. Do a smoke test across multiple sites and make sure that there are no regressions.

_Simple_
- Apply D59179-code to your WP.com sandbox.
- Sandbox the API, a domain-only site, a P2 site, and a simple site.
- Go to https://wordpress.com and switch any of the sandboxed sites.
- Check the menu under different scenarios (Calypso, WP Admin, toggle on/off) and make sure there are no regressions.
- Repeat with the rest of the sandboxed sites.

_Atomic_
- Install Jetpack Beta on an Atomic site and switch to the branch of this PR.
- Go to https://wordpress.com and switch to the WoA site.
- Check the menu under different scenarios (Calypso, WP Admin, toggle on/off) and make sure there are no regressions.

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		remove_submenu_page( 'https://wordpress.com/settings/general/' . $this->domain, 'ratings&action=options' );
+		remove_submenu_page( 'https://wordpress.com/settings/general/' . $this->domain, 'https://wordpress.com/hosting-config/' . $this->domain );
+		remove_submenu_page(
+			'https://wordpress.com/settings/general/' . $this->domain,
+			'https://wordpress.com/marketing/sharing-buttons/' . $this->domain
+		);
+
+		if ( $this->is_api_request ) {
+			// This menu page is only added on WP Admin.
+			add_menu_page( 'P2 Editor', 'P2 Editor', 'manage_options', 'p2editor', '', 'dashicons-admin-multisite' );
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php:66]: Should we use the `jetpack` textdomain and have these strings be translated?

File: projects/plugins/jetpack/modules/masterbar/admin-menu/load.php [hunk 2]
```diff
@@ -7,10 +7,56 @@
...
+		// Domain-only sites.
+		$blog_options   = get_blog_option( $blog_id, 'options' );
+		$is_domain_only = ! empty( $blog_options['is_domain_only'] );
+		if ( $is_domain_only ) {
+			require_once __DIR__ . '/class-domain-only-admin-menu.php';
+			return __NAMESPACE__ . '\Domain_Only_Admin_Menu';
+		}
+
+		// P2 sites.
+		if ( \WPForTeams\is_wpforteams_site( $blog_id ) ) {
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/load.php:35]: Do we need to call the `wpforteams` lib before we use `WPForTeams`?

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php (hunk 1 above)
Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php:66]: Seems we currently don't translate that in WP.com (see `/wp-content/plugins/p2-editor/classes/p2-editor-admin.php`) but not sure if it's intentional. cc @Automattic/lighthouse for a confidence check.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/load.php (hunk 2 above)
Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/load.php:36]: Yup, good point. Updated in 77a6c8c

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php (hunk 1 above)
Comment (lezama) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-p2-admin-menu.php:66]: not intentional :)

File: projects/plugins/jetpack/modules/masterbar/admin-menu/load.php [hunk 3]
```diff
@@ -7,10 +7,57 @@
...
+	// WordPress.com Simple sites.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		$blog_id = get_current_blog_id();
+
+		// Domain-only sites.
+		$blog_options   = get_blog_option( $blog_id, 'options' );
+		$is_domain_only = ! empty( $blog_options['is_domain_only'] );
+		if ( $is_domain_only ) {
+			require_once __DIR__ . '/class-domain-only-admin-menu.php';
+			return __NAMESPACE__ . '\Domain_Only_Admin_Menu';
```
Comment (BogdanUngureanu) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/load.php:31]: We can do something like `return Domain_Only_Admin_Menu::class;` to get the FQCN. IDEs offer support for that and makes the navigation a lot easier. :)

Comment (mmtr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/load.php]: Nice, much cleaner that way! Done in https://github.com/Automattic/jetpack/pull/19263/commits/32a8e4b1e4a691acea4ba8f772c696add52a0475

**Comment by lamosty** [comment]
Nice work!

Heads up, we've just deployed D59423-code which refactors some of the P2 Workspace parts which are also used in this PR. Could you please rebase?

**Comment by mmtr** [comment]
Thanks for the heads up @lamosty! I added them in the last commit.

**Review by BogdanUngureanu — APPROVED** [review]
LGTM!

**Review by jeherve — APPROVED** [review]
This looks good to me. 🚢

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.6` in b9bb7f3bfe00fbb64453a428d951c976b8f491fc

**Comment by mmtr** [comment]
Deployed to WP.com in r224037-wpcom

**Pull request merged**